docs: add sc docs (#4183)
aws/copilot-cli: site/content/docs/developing/svc-to-svc-communication.en.md

# Service-to-Service Communication

## Service Connect

!!! info

>     This feature is supported since Copilot `v1.24`, which provides more integrated service-to-service experience compared to [Service Discovery](#service-discovery).

Reviewer: what about https://github.com/aws/copilot-cli/blob/mainline/site/content/docs/developing/addons/package.en.md#L1 🥺

> ECS Service Connect enables your client services connect to their dependencies in a load-balanced and resilient fashion, simplifying the way of exposing those dependencies to their clients. With Service Connect in Copilot, each service you create is given a private alias by default: `http://<your service name>`.

Reviewer 2: It would be cool if we can include a link to the ECS docs here, maybe we can ask them what the URL will be?
```suggestion
[ECS Service Connect](some-ecs-doc-url) enables your client services connect to their dependencies in a load-balanced and resilient fashion, simplifying the way of exposing those dependencies to their clients. With Service Connect in Copilot, each service you create is given a private alias by default: `http://<your service name>`.
```

Reviewer 2: There might be a word that's missing here 😅 
```suggestion
ECS Service Connect enables a client service to connect to its downstream services in a load-balanced and resilient fashion. Furthermore, it simplifies the way of exposing a service to its clients by specifying friendly aliases. With Service Connect in Copilot, each service you create is given the following private alias by default: `http://<your service name>`.
```

### How do I use Service Connect?
Imagine we have an app called `kudos` and two services: `api` and `front-end`, deployed in the same environment. In order to use Service Connect, both services' manifests need to have:

```yaml
network:
  connect: true # Defaults to "false"
```

After deploying both services, they should be able to talk to each other using the default Service Connect endpoint, which is the same as its service name. For example, `front-end` service can simply call `http://api`.

> ```go

Reviewer: ❤️

// Calling the "api" service from the "front-end" service.
resp, err := http.Get("http://api/")

> ```

Reviewer 2: This reminds me (in a separate PR), I wonder if we should a recommended action after `copilot deploy` saying:
```
Your service is accessible within the VPC at: fmt.Sprintf(http://%s:%s/, name, port)
```

Author: I think I added that like

```
Recommended follow-up action:
  - You can access your service at redis-server:6379 with service connect.
```

### Upgrading from Service Discovery

> Prior to v1.24, Copilot enabled private service-to-service communication with Service Discovery. If you are already using Service Discovery and want to avoid any code changes, you can configure [`network.connect.alias`](../manifest/lb-web-service.en.md#network-connect-alias) field so that the Service Connect uses the same alias as Service Discovery. And if **both** client and caller have Service Connect enabled, they'll connect via Service Connect instead of Service Discovery. For example, in the manifest of the `api` service we have

Reviewer 2: isn't client and caller the same thing?
```suggestion
Prior to v1.24, Copilot enabled private service-to-service communication with Service Discovery. If you are already using Service Discovery and want to avoid any code changes, you can configure [`network.connect.alias`](../manifest/lb-web-service.en.md#network-connect-alias) field so that the Service Connect uses the same alias as Service Discovery. And if **both**, the service and its client, have Service Connect enabled, they'll connect via Service Connect instead of Service Discovery. For example, in the manifest of the `api` service we have
```

```yaml
network:
  connect:
    alias: ${COPILOT_SERVICE_NAME}.${COPILOT_ENVIRONMENT_NAME}.${COPILOT_APPLICATION_NAME}.local
```

> and `front-end` can keep using the same endpoint to make API calls via Service Connect instead of Service Discovery to leverage the benefits of load balancing and additional resiliency.

Reviewer 2: ```suggestion
and `front-end` also has the same setting. Then, they can keep using the same endpoint to make API calls via Service Connect instead of Service Discovery to leverage the benefits of load balancing and additional resiliency.
```
